metricbeat/module/prometheus/collector/_meta/testdata/histogram.plain-expected.json

```diff
@@ -0,0 +1,71 @@
+[
+    {
+        "event": {
+            "dataset": "prometheus.collector",
+            "duration": 115000,
+            "module": "prometheus"
+        },
+        "metricset": {
+            "name": "collector"
+        },
+        "prometheus": {
+            "labels": {
+                "code": "200",
+                "le": "50"
+            },
+            "metrics": {
+                "hash_seconds_bucket": 3
+            }
+        },
+        "service": {
+            "address": "127.0.0.1:55555",
+            "type": "prometheus"
+        }
+    },
+    {
+        "event": {
+            "dataset": "prometheus.collector",
+            "duration": 115000,
+            "module": "prometheus"
+        },
+        "metricset": {
+            "name": "collector"
+        },
+        "prometheus": {
+            "labels": {
+                "code": "200",
+                "le": "10"
+            },
+            "metrics": {
+                "hash_seconds_bucket": 2
+            }
+        },
+        "service": {
+            "address": "127.0.0.1:55555",
+            "type": "prometheus"
+        }
+    },
+    {
+        "event": {
+            "dataset": "prometheus.collector",
+            "duration": 115000,
+            "module": "prometheus"
+        },
+        "metricset": {
+            "name": "collector"
+        },
+        "prometheus": {
+            "labels": {
+                "code": "200"
+            },
+            "metrics": {
+                "hash_seconds_count": 0,
```

Comment:
Can anyone help me understand why this metric exists please? I checked regenerating this json file without the change to ignore NaN, this metric still got generated. Is this something related to histogram?

Reply:
yes, check https://github.com/elastic/beats/blob/master/metricbeat/module/prometheus/collector/data.go#L106, I guess numbers are 0 here because of NaN

Reply:
@exekias ahh it's the `sum` and `count` from histogram metrics. I did not specify a `sum` or `count` here in `histogram.plain` and that's why its showing 0.